docs/config.md

```diff
@@ -16,74 +18,88 @@ metadata:
+| AVP_MOUNT_PATH             | Vault Auth Mount PATH                               | Optional. Defaults to the appropriate path based on `AUTH_TYPE` (i.e, `auth/approle` for AppRole authentication, `auth/github` for Github, `auth/kubernetes` for Kubernetes) |
+| AVP_K8S_MOUNT_PATH         | Kuberentes Auth Mount PATH                          | Optional for `AUTH_TYPE` of `k8s` defaults to `auth/kubernetes`. Takes precedence over `$AVP_MOUNT_PATH`                                                                     |
+| AVP_K8S_ROLE               | Kuberentes Auth Role                                | Required with `AUTH_TYPE` of `k8s`                                                                                                                                           |
+| AVP_K8S_TOKEN_PATH         | Path to JWT for Kubernetes Auth                     | Optional for `AUTH_TYPE` of `k8s` defaults to `/var/run/secrets/kubernetes.io/serviceaccount/token`                                                                          |
+| AVP_IBM_API_KEY            | IBM Cloud IAM API Key                               | Required with `TYPE` of `ibmsecretsmanager`                                                                                                                                  |
+| AVP_IBM_INSTANCE_URL       | Endpoint URL for IBM Cloud Secrets Manager instance | If absent, fall back to `$VAULT_ADDR`                                                                                                                                        |
+| AWS_REGION                 | AWS Secrets Manager Region                          | Only valid with `TYPE` `awssecretsmanager`                                                                                                                                   |
+| AVP_YCL_SERVICE_ACCOUNT_ID | Yandex Cloud Lockbox service account ID             | Required with `TYPE` of `yandexcloudlockbox`                                                                                                                                 |
+| AVP_YCL_KEY_ID             | Yandex Cloud Lockbox service account Key ID         | Required with `TYPE` of `yandexcloudlockbox`                                                                                                                                 |
+| AVP_YCL_PRIVATE_KEY        | Yandex Cloud Lockbox service account private key    | Required with `TYPE` of `yandexcloudlockbox`                                                                                                                                 |
+| OP_CONNECT_TOKEN           | 1Password Connect access token                      | Required with `TYPE` of `1passwordconnect`                                                                                                                                   |
```

Comment:
If `OP_CONNECT` prefixed env vars are what the SDK expects I don't think we need to explicitly call them out here, the note above this table should point users to the right place:

> We support all the backend specific environment variables each backend's SDK will accept (e.g, `VAULT_NAMESPACE`, `AWS_REGION`, etc). Refer to the [specific backend's documentation](./backends) for details.

Otherwise, we should keep them here